docs/visual-basic/programming-guide/language-features/data-types/tuples.md

```diff
@@ -0,0 +1,153 @@
+
+[!code-vb[Instantiate](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple1.vb#3)]
+
+## Instantiating and using a named tuple
+
+Rather than using default names for a tuple's fields, you can instantiate a *named tuple* by assigning your own names to the tuple's elements. The tuple's fields can then be accessed by their assigned names *or* by their default names. The following example instantiates the same 3-tuple as previously, except that it explicitly names the first field `EventDate`, the second `Name`, and the third `IsHoliday`. It then displays the field values, modifies them, and displays the field values again.
+
+[!code-vb[Instantiate](../../../../../samples\snippets\visualbasic\programming-guide\language-features\data-types/tuple1.vb#4)]
+
+## Tuples versus structures
+
+A Visual Basic tuple is a value type that is an instance of one of the a **System.ValueTuple** generic types. For example, the `holiday` tuple defined in the previous example is an instance of the <xref:System.ValueTuple%603> structure. It is designed to be a lightweight container for data. Since the tuple aims to make it easy to create an object with multiple data items, it lacks some of the features that a custom structure might have. These include:
+
+- Customer members. You cannot define your own properties, methods, or events for a tuple.
+
+- Validation. You cannot validate the data assigned to fields.
+
+- Mutability. Visual Basic tuples are mutable. In contrast, a custom structure allows you to control whether an instance is mutable or immutable.
+
+If custom members, property and field validation, or immutability are important, you should use the Visual Basic [Structure](../../../language-reference/statements/structure-statement.md) statement to define a custom value type.
+
+A Visual Basic tuple does inherit the members of its **ValueTuple** type. In addition to its fields, these include the following methods:
+
+| Member | Description |
+| ---|---|
+| CompareTo | Compares the current tuple to another tuple with the same number of elements. |
+| Equals | Determines whether the current tuple is equal to another tuple or object. |
+| GetHashCode | Calculates the hash code for the current instance. |
+| ToString | Returns the string representation of this tuple, which takes the form `(Item1, Item2...)`, where `Item1` and `Item2` represent the values of the tuple's fields. |
+
+In addition, the **ValueTuple** types implement <xref:System.Collections.IStructuralComparable> and <xref:System.Collections.IStructurealEquatable> interfaces, which allow you to define customer comparers.
+
+## Assignment and tuples
+
+Visual Basic supports assignment between tuple types that have the same number of fields and the same type for each field. Those types must be exact compile-time matches. Other conversions are not considered for assignments. Let's look at the kinds of assignments that are allowed between tuple types.
```

Comment:
The second sentence is incorrect. The types of each field don't have to be the same. There must be an implicit conversion from the field type in the right-hand-side tuple to the field type in the left-hand-side tuple.  As a trivial example:

```vb
        Dim longs = (19L, 23L)
        Dim ints = (19, 23)
        longs = ints
```